lib/hashdiff.rb

```diff
@@ -1,10 +1,14 @@
 # frozen_string_literal: true
 
-require_relative 'hashdiff/util'
-require_relative 'hashdiff/compare_hashes'
-require_relative 'hashdiff/lcs'
-require_relative 'hashdiff/lcs_compare_arrays'
-require_relative 'hashdiff/linear_compare_array'
-require_relative 'hashdiff/diff'
-require_relative 'hashdiff/patch'
-require_relative 'hashdiff/version'
+require 'hashdiff/util'
```

Comment:
Why did you change it from `require_relative` to `require`?

Reply:
@krzysiek1507 it's in my commit message

Reply:
@jfelchner could you please share more resources about this? Blog posts, issues etc.

Reply:
Here's just one: https://github.com/rails/rails/issues/29846#issuecomment-338442315

Regardless there are many, many edge cases for using `require_relative` and it should never be used in a library.  Using it in a project is technically ok, although I never do that either.

Reply:
Also the `/lib` path is added to the relative load directory, so these absolute paths will always work inside this gem (As they do inside most gems that are built in the normal way)

Reply:
Ok, let's change it. I've never seen such issue before.